src/datamigration/azext_datamigration/manual/_params.py

```diff
@@ -63,6 +63,17 @@ def load_arguments(self, _):
         c.argument('aad_domain_name', type=str, help='Required if Windows logins are included in the list of logins to be migrated.')
         c.argument('config_file_path', type=file_type, completer=FilesCompleter(), help='Path of the ConfigFile')
 
+    with self.argument_context('datamigration tde-migration') as c:
+        c.argument('source_sql_connection_string', type=str, help='Connection string for the source SQL instance, using the formal connection string format.')
+        c.argument('target_subscription_id', type=str, help='Subscription Id of the target Azure SQL server.')
+        c.argument('target_resource_group_name', type=str, help='Resource group name of the target Azure SQL server.')
+        c.argument('target_managed_instance_name', type=str, help='Name of the Azure SQL Server.')
+        c.argument('network_share_path', type=str, help='Network share path.')
+        c.argument('network_share_domain', type=str, help='Network share domain.')
+        c.argument('network_share_user_name', type=str, help='Network share user name.')
+        c.argument('network_share_password', type=str, help='Network share password.')
```

Comment:
When I made changes to Azure PowerShell repo, they suggested making network_share_password and source_sql_connection_string SecureStrings.

Since python is parsing these arguments, not sure how that would work here.  Seems like other cmdlets just use "type=str" for passwords.

What is the guidance here?  Should I keep it as "type=str"?

Reply:
Hi @smartura, it seems your commands are rely on management-plane sdk. You should try our codegen tool to generate those commands. It supports prompt input for password kinds arguments. You can reference these two PRs https://github.com/Azure/aaz-dev-tools/pull/238 and https://github.com/Azure/azure-cli/pull/26134

Reply:
This command is manual, not generated.  Is there an example of a manual command using prompt arguments?

Reply:
You can use prompt_pass function. Using aaz-dev-tools is highly recommended.